### ISSUE TITLE
read -t 0 does not require -r

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#1709]
#### For bugs
- Rule Id SC2162
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
read -t 0 && IFS='' read -d '' -r std_in

```

#### Here's what shellcheck currently says:
^-- SC2162: read without -r will mangle backslashes.

#### Here's what I wanted or expected to see:
no report, it is ok

**Comment by koalaman** [comment]
Thanks! I additionally tweaked it to allow things like `read -u 3 -t 0`

**Comment by koalaman** [comment]
Err, I mean: Fixed by @iboss-ptk. Thanks!
